Sources/SocksCore/TCPSocket.swift

```diff
@@ -39,22 +39,29 @@ public protocol TCPReadableSocket: TCPSocket { }
 extension TCPReadableSocket {
 
     public func recv(maxBytes: Int = BufferCapacity) throws -> [UInt8] {
+    	var out = Array<UInt8>()
         let data = Bytes(capacity: maxBytes)
         let flags: Int32 = 0 //FIXME: allow setting flags with a Swift enum
         let receivedBytes = socket_recv(self.descriptor, data.rawBytes, data.capacity, flags)
-        guard receivedBytes > -1 else { throw SocksError(.readFailed) }
-        
-        guard receivedBytes > 0 else {
+        guard receivedBytes > -1 else {
+	    if errno == 104 { // closed by peer, need to close this side too!
```

Comment:
Brett:  I was having some issues getting the right syntax to get the values from ERRNO symbolically in the code.  I don't like them, so I am glad you called me on it.